timesketch/api/v1/resources/sigma.py

```diff
@@ -89,4 +95,73 @@ def get(self, rule_uuid):
             abort(
                 HTTP_STATUS_CODE_NOT_FOUND, 'No sigma rule found with this ID.')
 
-        return return_rule
+        meta = {'current_user': current_user.username,
+                'rules_count': len(sigma_rules)}
+        return jsonify({'objects': return_rule, 'meta': meta})
+
+class SigmaByTextResource(resources.ResourceMixin, Resource):
+    """Resource to get a Sigma rule by text."""
+
+    @login_required
+    def post(self):
+        """Handles POST request to the resource.
+
+        Returns:
+            JSON sigma rule
+        """
+
+        form = request.json
+        if not form:
+            form = request.data
+
+        action = form.get('action', '')
+        
+        if action != 'post':
+            return abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                'Action needs to be "post"')
+        if action == 'post':
+            content = form.get('content')
+            if not content:
+                return abort(
+                    HTTP_STATUS_CODE_BAD_REQUEST,
+                    'Missing values from the request.')
+
+        try:
+            sigma_rule = ts_sigma_lib.get_sigma_rule_by_text(content)
+
+        except ValueError:
+            logger.error('Parsing error',
+                         exc_info=True)
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                'Error unable to parse the provided Sigma rule')
```

Comment:
do you want to contain some of the error report here, or is this sufficient?

Reply:
well that is kind of the else case where it is really unclear what was wrong... all the other cases are more granular, this one is the else tree of things